src/java/org/apache/cassandra/db/filter/RowFilter.java

```diff
@@ -1077,7 +1106,54 @@ public int hashCode()
         @Override
         protected Kind kind()
         {
-            return Kind.MAP_EQUALITY;
+            return Kind.MAP_COMPARISON;
+        }
+
+        /**
+         * Get the lower bound for this expression. When the expression is EQ, GT, or GTE, the lower bound is the
+         * expression itself. When the expression is LT or LTE, the lower bound is the map's key becuase
+         * {@link ByteBuffer} comparisons will work correctly.
+         * @return the lower bound for this expression.
+         */
+        public ByteBuffer getLowerBound()
+        {
+            switch (operator) {
+                case EQ:
+                case GT:
+                case GTE:
+                    return this.getIndexValue();
+                case LT:
+                case LTE:
+                    return CompositeType.build(ByteBufferAccessor.instance, key);
+                default:
+                    throw new AssertionError();
+            }
+        }
+
+        /**
+         * Get the upper bound for this expression. When the expression is EQ, LT, or LTE, the upper bound is the
+         * expression itself. When the expression is GT or GTE, the upper bound is the map's key with the last byte
+         * set to 1 so that {@link ByteBuffer} comparisons will work correctly.
+         * @return the upper bound for this express
+         */
+        public ByteBuffer getUpperBound()
+        {
+            switch (operator) {
+                case GT:
+                case GTE:
+                    // TODO this is a leak in the abstraction. Where does this go? It works because we know the way the
+                    // bytes are always indexed with a terminating 0, we can guarantee our
+                    // upper bound is inclusive of all values by setting the last byte to 1
+                    var result = CompositeType.build(ByteBufferAccessor.instance, key);
+                    result.put(result.limit() - 1, (byte) 1);
+                    return result;
```

Comment:
I wasn't sure about this, but I did manage to find proof that this is a correct usage (though it definitely needs to be encapsulated better)

https://github.com/datastax/cassandra/blob/53db16b679adb58f52b67fa34ea26831dac853ae/src/java/org/apache/cassandra/db/marshal/CompositeType.java#L48-L59